set_effort_analysis/github_contributions_fy_2023.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # DBMI SET GitHub Contributions - Fiscal Year 2023
+# 
+# Note: please set environment variables SET_EFFORT_GH_USER with your GitHub username and SET_EFFORT_GH_TOKEN with your GitHub token in order to use this notebook.
+
+# In[1]:
+
+
+import json
+import os
+from datetime import datetime
+
+import requests
+
+
+# In[2]:
+
+
+# Provide your GitHub username and personal access token for authentication
+username = os.environ.get("SET_EFFORT_GH_USER")
+token = os.environ.get("SET_EFFORT_GH_TOKEN")
+
+date_start = datetime.strptime("2022-07-01", "%Y-%m-%d")
+date_end = datetime.strptime("2023-06-30", "%Y-%m-%d")
+
+# specify set members as github usernames
+set_members = ["vincerubinetti", "falquaddoomi", "d33bs"]
+
+# Define github organization names
+orgs = [
+    "cu-dbmi",
+    "cytomining",
+    "wayscience",
+    "JRaviLab",
+    "krishnanlab",
+    "greenelab",
+    "manubot",
+    "CCPM-TIS",
+    "monarch-initiative",
+    "blekhmanlab",
+    "tis-lab",
+    "hetio",
+    "biothings",
+]
+
+# Create a re-usable session object with the user creds in-built
+gh_session = requests.Session()
+gh_session.auth = (username, token)
+
+
+# In[3]:
+
+
+def within_time_range(
+    date_to_check: datetime,
+    date_start: datetime = date_start,
+    date_end: datetime = date_end,
+) -> bool:
+    """
+    Checks whether given date falls within range of date_start and date_end
+    """
+
+    return date_start <= date_to_check <= date_end
+
+
+# In[4]:
+
+
+org_repos = []
+
+for org in orgs:
+    # fetch the list of repositories in the organizations, appending each
+    org_repos += json.loads(
+        gh_session.get(f"https://api.github.com/orgs/{org}/repos").text
+    )
+
+
+# In[5]:
+
+
+# set parameters for gathering data
+params = {
+    # get issues with all states
+    "state": "all",
+    # get only the first page of 400 results
+    "page": "1",
+    "per_page": 400,
```

Comment:
I think the max value here is 100?
https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28

So for certain orgs that have a ton of repos, like the greenelab, you might be missing some. You might have to loop through the pagination here.

Reply:
Looks like greenelab, monarch-init, and krishnanlab are the only ones that go over 100. And the first two go over significantly.

Reply:
Thanks for this comment, great point! This has been addressed using PyGithub's built-in pagination.